### PR TITLE
Xwfm  gateway deployment ansible roles [xwfm]

### PR DESCRIPTION
**Pull request by bennojoy**
## Summary

this PR has the ansible roles that allow us to deploy a new xwfm gateway in a centos box. it has flexibility to deploy the gateway in multiple modes like passhthrough/ internal dhcp etc.

## Test Plan

tested by deploying the gateways in multple centos boxes.

### CODE REVIEW

File: xwf/gateway/deploy/roles/pkgrepo/files/jfrog.pub [hunk 1]
```diff
@@ -0,0 +1,30 @@
+-----BEGIN PGP PUBLIC KEY BLOCK-----
```
Comment (tmdzk) [inline xwf/gateway/deploy/roles/pkgrepo/files/jfrog.pub]: There is a link to directly get the public [gpg key](https://facebookconnectivity.jfrog.io/artifactory/api/gpg/key/public ) can you use it in order to not host it in the codebase and dynamically pull it ? 

`wget -qO - https://facebookconnectivity.jfrog.io/artifactory/api/gpg/key/public | apt-key add -` something like that should do the trick (if wget and gnupg* is installed)

File: xwf/gateway/deploy/roles/pkgrepo/tasks/debian.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  apt_key:
+    url: http://packages.magma.etagecom.io/pubkey.gpg
+    state: present
+    id: 2E97854368E14ED8FB6F80903E838E02450E78EA
+
+- name: Add apt-transport-https
+  apt: pkg=apt-transport-https state=present update_cache=yes
+  #  when: preburn
+
+- name: Add JFrog key
```
Comment (tmdzk) [inline xwf/gateway/deploy/roles/pkgrepo/tasks/debian.yml:23]: can you do the same as the Add GPG key for magma repository for the jfrog one ? see comment above

File: xwf/gateway/deploy/roles/pkgrepo/tasks/redhat.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+################################################################################
+
+- name: Add GPG key for magma repository
+  rpm_key:
+    key: https://facebookconnectivity.jfrog.io/artifactory/{{ repo }}/{{ distribution }}/7/repodata/repomd.xml.key
+    state: present
+
+- name: Add JFrog key
```
Comment (tmdzk) [inline xwf/gateway/deploy/roles/pkgrepo/tasks/redhat.yml:19]: Can be removed when you changed the way to get the pub key

File: xwf/gateway/deploy/roles/provision/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+- name: install python modules
+  shell: pip3 install jsonpickle
+
+- name: touch snowflake
```
Comment (tmdzk) [inline xwf/gateway/deploy/roles/provision/tasks/main.yml]: Why do you need to touch the snowflake file ?

File: xwf/gateway/deploy/roles/provision/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  shell: touch /etc/snowflake
+
+- name: retrieve hwid
```
Comment (tmdzk) [inline xwf/gateway/deploy/roles/provision/tasks/main.yml:17]: Hardware id can be found in `/etc/snowflake`

File: xwf/gateway/deploy/roles/pkgrepo/files/jfrog.pub (hunk 1 above)
Comment (bennojoy) [reply xwf/gateway/deploy/roles/pkgrepo/files/jfrog.pub]: its done

File: xwf/gateway/deploy/roles/pkgrepo/tasks/redhat.yml (hunk 3 above)
Comment (bennojoy) [reply xwf/gateway/deploy/roles/pkgrepo/tasks/redhat.yml:19]: Done.

File: xwf/gateway/deploy/roles/provision/tasks/main.yml (hunk 5 above)
Comment (bennojoy) [reply xwf/gateway/deploy/roles/provision/tasks/main.yml:17]: done

**Review by tmdzk — APPROVED** [review]

**Pull request merged**